pthread_support.c

```diff
@@ -1510,12 +1510,7 @@ fork_child_proc(void)
 #    endif
 #    ifdef PARALLEL_MARK
   if (GC_parallel) {
-#      if defined(THREAD_SANITIZER) && defined(GC_ASSERTIONS) \
```

Comment:
Will it be OK if the existing pthread_mutex_unlock/GC_release_mark_lock call is left as-is, but after it we call pthread_mutex_init?

Reply:
It will still error, but since the error isn't checked at the moment except in assert builds, it isn't causing any problems. Ideally it should not be unlocked at all but reinitialized immediately.

Reply:
> Ideally it should not be unlocked at all but reinitialized immediately.

The manual (https://man7.org/linux/man-pages/man3/pthread_mutex_destroy.3p.html) says the opposite:
- Attempting to destroy a locked mutex, or a mutex that another thread is attempting to lock, ..., results in undefined behavior
- Attempting to initialize an already initialized mutex results in undefined behavior

> It will still error, but since the error isn't checked at the moment except in assert builds, it isn't causing any problems.

Right, the result is cast to void intentionally (both for `mark_mutex` and `GC_allocate_ml`).
I will remove checking in the assertion, thank you!